### ISSUE TITLE
importer mapping for .gdx files does not read variable type data

### ISSUE THREAD

**Issue by nnhjy** [Spine-project/Spine-Toolbox#1741]
When reading a .gdx file, import displays values for only the `parameter` type data in the mapping but nothing for `variable` type data. An error occurs with the message "Could not get data from source: 'NoneType' object has no attribute 'domain'". Please use [sample.zip](https://github.com/Spine-project/Spine-Toolbox/files/9262723/sample.zip) to replicate the issue.

**Comment by nnhjy** [comment]
The problem may be also due to a lack of support on `GAMSVariable` in the `GDX2py` package. I also create an [issue](https://github.com/erkkar/GDX2py/issues/6#issue-1330146267) in its repository.

**Comment by soininen** [comment]
> The problem may be also due to a lack of support on GAMSVariable in the GDX2py package.

Indeed. I improved the error message somewhat but before GDX2py supports variables there isn't much we can do here.

**Comment by nnhjy** [comment]
> > The problem may be also due to a lack of support on GAMSVariable in the GDX2py package.
> 
> Indeed. I improved the error message somewhat but before GDX2py supports variables there isn't much we can do here.

Thanks! Let's hope Erkka would have some time to spend on that.

**Comment by erkkar** [comment]
Hi, indeed GDX2py does not have a class for GAMS variables, see https://github.com/erkkar/GDX2py/blob/master/gdx2py/gams.py. I’ve probably skipped those because there was no need at that time. I’m guessing it would not be a big thing, though. Basic idea would be like `GAMSParameter` but the `data` attribute has to be different as variables have a level value, upper and lower limits as well as a scale value for each index. Two options:
1. `data` is a mapping from tuple --> list or dict
2. There are separate `level`, `upper` etc. attributes for each of these.
If it’s easy, I might do something for it in the near future. You’re of course also free to submit a pull request ;)
